crates/solvers/src/infra/oracle/oneinch.rs

```diff
@@ -0,0 +1,58 @@
+use crate::domain::{auction, eth};
+
+/// 1Inch on-chain spot price aggregator.
+pub struct OneInch {
+    oracle: contracts::OneInchOffchainOracle,
+}
+
+pub struct Config {
+    /// The Ethereum RPC URL to use.
+    pub ethrpc: reqwest::Url,
+
+    /// The address of the `OffchainOracle` contract. Specify `None` to use the
+    /// default contract address for the connected chain.
+    pub oracle: Option<eth::ContractAddress>,
+}
+
+impl OneInch {
+    pub async fn new(config: Config) -> Result<Self, Error> {
+        let web3 = ethrpc::web3(module_path!(), &config.ethrpc);
```

Comment:
Enabling the `trace` logs of `ethrpc` already puts a lot of pressure on elastic search. I think having a short name here would actually make a difference.
```suggestion
        let web3 = ethrpc::web3("OneInchOracle", &config.ethrpc);
```